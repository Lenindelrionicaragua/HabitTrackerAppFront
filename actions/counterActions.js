import AsyncStorage from "@react-native-async-storage/async-storage";
import { createSelector } from "reselect";

export const SET_INITIAL_TIME = "SET_INITIAL_TIME";
export const SET_REMAINING_TIME = "SET_REMAINING_TIME";
export const SET_ELAPSED_TIME = "SET_ELAPSED_TIME";
export const SET_TIME_COMPLETED = "SET_TIME_COMPLETED";
export const SET_IS_RUNNING = "SET_IS_RUNNING";
export const SET_FIRST_RUN = "SET_FIRST_RUN";
export const SET_START_TIME_REF = "SET_START_TIME_REF";

export const SET_RESET_BUTTON_LABEL = "SET_RESET_BUTTON_LABEL";
export const SET_INFO_TEXT = "SET_INFO_TEXT";
export const SET_RESET_CLICKS = "SET_RESET_CLICKS";
export const SET_RESET_TIMEOUTS_IDS = "SET_RESET_TIMEOUTS_IDS";
export const SET_HAS_STARTED = "SET_HAS_STARTED";
export const SET_CIRCLE_COLOR = "SET_CIRCLE_COLOR";
export const SET_INNER_CIRCLE_COLOR = "SET_INNER_CIRCLE_COLOR";
export const SET_BUTTONS_DISABLED = "SET_BUTTONS_DISABLED";
export const SET_SAVE_TIME_BUTTON_LABEL = "SET_SAVE_TIME_BUTTON_LABEL";

export const SET_HABIT_CATEGORIES = "SET_HABIT_CATEGORIES";
export const SET_HABIT_CATEGORY_INDEX = "SET_HABIT_CATEGORY_INDEX";
export const RESET_HABIT_CATEGORIES = "RESET_HABIT_CATEGORIES";

export const SET_MONTHLY_STATS = "SET_MONTHLY_STATS";
export const CLEAR_MONTHLY_STATS = "CLEAR_MONTHLY_STATS";

export const SET_PAUSE_TIME = "SET_PAUSE_TIME";

export const SET_ACTIVE_SCREEN = "SET_ACTIVE_SCREEN";

export const setInitialTime = newInitialTime => {
  return {
    type: SET_INITIAL_TIME,
    payload: newInitialTime
  };
};

export const setRemainingTime = newRemainingTime => {
  return {
    type: SET_REMAINING_TIME,
    payload: newRemainingTime
  };
};

export const setElapsedTime = newElapsedTime => {
  return {
    type: SET_ELAPSED_TIME,
    payload: newElapsedTime
  };
};

export const setTimeCompleted = newTimeCompleted => {
  return {
    type: SET_TIME_COMPLETED,
    payload: newTimeCompleted
  };
};

export const setIsRunning = newIsRunning => {
  return {
    type: SET_IS_RUNNING,
    payload: newIsRunning
  };
};

export const setFirstRun = newFirstRun => {
  return {
    type: SET_FIRST_RUN,
    payload: newFirstRun
  };
};

export const setResetButtonLabel = newResetButtonLabel => {
  return {
    type: SET_RESET_BUTTON_LABEL,
    payload: newResetButtonLabel
  };
};

export const setInfoText = newInfoText => {
  return {
    type: SET_INFO_TEXT,
    payload: newInfoText
  };
};

export const setResetClicks = newResetClicks => {
  return {
    type: SET_RESET_CLICKS,
    payload: newResetClicks
  };
};

export const setResetTimeoutsIds = timeoutsIds => {
  return {
    type: SET_RESET_TIMEOUTS_IDS,
    payload: timeoutsIds
  };
};

export const setHasStarted = newHasStarted => {
  return {
    type: SET_HAS_STARTED,
    payload: newHasStarted
  };
};

export const setCircleColor = newCircleColor => {
  return {
    type: SET_CIRCLE_COLOR,
    payload: newCircleColor
  };
};

export const setInnerCircleColor = newInnerCircleColor => {
  return {
    type: SET_INNER_CIRCLE_COLOR,
    payload: newInnerCircleColor
  };
};

export const setPauseTime = pauseTime => {
  return {
    type: SET_PAUSE_TIME,
    payload: pauseTime
  };
};

export const setSaveTimeButtonLabel = saveTimeButtonLabel => {
  return {
    type: SET_SAVE_TIME_BUTTON_LABEL,
    payload: saveTimeButtonLabel
  };
};

export const setButtonsDisabled = buttonsDisabled => {
  return {
    type: SET_BUTTONS_DISABLED,
    payload: buttonsDisabled
  };
};

export const setHabitCategoryIndex = habitCategoryIndex => {
  return {
    type: SET_HABIT_CATEGORY_INDEX,
    payload: habitCategoryIndex
  };
};

// Action to load habit categories from AsyncStorage
export const loadHabitCategories = () => {
  return async dispatch => {
    try {
      const storedCategories = await AsyncStorage.getItem("habitCategories");
      if (storedCategories) {
        const parsedCategories = JSON.parse(storedCategories);
        dispatch({
          type: SET_HABIT_CATEGORIES,
          payload: parsedCategories
        });
      }
    } catch (error) {
      console.error("Error loading categories from AsyncStorage:", error);
    }
  };
};

export const setHabitCategories = habitCategories => ({
  type: SET_HABIT_CATEGORIES,
  payload: habitCategories
});

export const resetHabitCategories = () => ({
  type: RESET_HABIT_CATEGORIES
});

const selectMonthlyStats = state => state.monthlyStats;

export const selectMonthlyStatsWithFallback = createSelector(
  [selectMonthlyStats],
  stats => ({
    ...stats,
    series: stats.series.length ? stats.series : [1], // Example fallback value
    sliceColors: stats.sliceColors.length ? stats.sliceColors : ["#cccccc"] // Fallback color
  })
);

export const clearMonthlyStats = () => ({
  type: CLEAR_MONTHLY_STATS
});

export const setActiveScreen = screen => {
  return {
    type: SET_ACTIVE_SCREEN,
    payload: screen
  };
};
