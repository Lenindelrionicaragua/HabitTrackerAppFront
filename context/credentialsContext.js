import { createContext, useState } from "react";
import { logInfo } from "../util/logging";

export const CredentialsContext = createContext();

export const CredentialsProvider = ({ children }) => {
  const [storedCredentials, setStoredCredentials] = useState({});

  const updateCredentials = credentials => {
    setStoredCredentials(credentials);
  };

  return (
    <CredentialsContext.Provider
      value={{ storedCredentials, setStoredCredentials: updateCredentials }}
    >
      {children}
    </CredentialsContext.Provider>
  );
};
