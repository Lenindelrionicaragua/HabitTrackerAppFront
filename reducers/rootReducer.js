import { combineReducers } from "redux";
import counterReducer from "../reducers/counterReducer";

const rootReducer = combineReducers({
  counter: counterReducer
});

export default rootReducer;
