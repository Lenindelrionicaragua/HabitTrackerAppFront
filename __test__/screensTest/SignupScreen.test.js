import React from "react";
import renderer from "react-test-renderer";
import {
  render,
  fireEvent,
  act,
  waitFor,
  cleanup
} from "@testing-library/react-native";
import { Formik } from "formik";
import { StatusBar } from "react-native";
import { Provider } from "react-redux";
import { createStore } from "redux";
import SignupScreen from "../../screens/SignupScreen/SignupScreen";
import rootReducer from "../../reducers/rootReducer";
import { CredentialsContext } from "../../context/credentialsContext";
import { setActiveScreen } from "../../actions/counterActions";

// Mock navigation object
const mockNavigation = {
  navigate: jest.fn()
};

// Create a mock store with initial state
const initialState = {
  activeScreen: {
    activeScreen: "mockActiveScreen"
  }
};

const store = createStore(rootReducer, initialState);

// Function to render SignupScreen wrapped in Provider and CredentialsContext,
// passing the mockNavigation prop explicitly
const renderSignupScreen = () =>
  render(
    <Provider store={store}>
      <CredentialsContext.Provider
        value={{ storedCredentials: null, setStoredCredentials: jest.fn() }}>
        <SignupScreen navigation={mockNavigation} />
      </CredentialsContext.Provider>
    </Provider>
  );

const renderSignupScreenWithRenderer = () =>
  renderer.create(
    <Provider store={store}>
      <CredentialsContext.Provider
        value={{ storedCredentials: null, setStoredCredentials: jest.fn() }}>
        <SignupScreen navigation={mockNavigation} />
      </CredentialsContext.Provider>
    </Provider>
  );

let signupScreenRender;
let signupScreenRenderWithRenderer;

beforeEach(() => {
  signupScreenRender = renderSignupScreen();
  signupScreenRenderWithRenderer = renderSignupScreenWithRenderer();
});

// SignupScreen Tests
describe("SignupScreen", () => {
  afterEach(() => {
    cleanup();
  });

  test("Renders correctly the SignupScreen Component", () => {
    const signupScreenSnapshot = signupScreenRenderWithRenderer.toJSON();
    expect(signupScreenSnapshot).toMatchSnapshot();
  });

  test("Render StatusBar correctly", () => {
    const signupScreenInstance = signupScreenRenderWithRenderer.root;
    const statusBar = signupScreenInstance.findByType(StatusBar);
    expect(statusBar).toBeTruthy();
  });

  test("Render UI components correctly", () => {
    const { getByTestId } = signupScreenRender;

    expect(getByTestId("signup-styled-container")).toBeTruthy();
    expect(getByTestId("inner-container")).toBeTruthy();
    expect(getByTestId("signup-page-title")).toBeTruthy();
    expect(getByTestId("signup-page-sub-title")).toBeTruthy();
  });

  test("PageTitle should render a string of letters, numbers, or spaces", () => {
    const { getByTestId } = signupScreenRender;
    const pageTitleComponent = getByTestId("signup-page-title");
    const textContent = pageTitleComponent.props.children.toString();
    expect(textContent).toMatch("Habit Tracker");
  });

  test("SubTitle should render a string of letters, numbers or spaces", () => {
    const { getByTestId } = signupScreenRender;
    const subTitleComponent = getByTestId("signup-page-sub-title");
    const textContent = subTitleComponent.props.children.toString();
    expect(textContent).toMatch("Account Sign Up");
  });
});

// Formik Integration Tests
describe("Formik Integration Tests", () => {
  let formikComponent;

  beforeEach(() => {
    const signupScreenInstance = signupScreenRenderWithRenderer.root;
    formikComponent = signupScreenInstance.findByType(Formik);
  });

  afterEach(() => {
    cleanup();
  });

  test("Render a Formik component", () => {
    expect(formikComponent).toBeTruthy();
  });

  test("Should have initialValues", () => {
    expect(formikComponent.props.initialValues).toEqual({
      name: "",
      email: "",
      dateOfBirth: "",
      password: "",
      confirmPassword: ""
    });
  });

  test("Should have a function as a child", () => {
    expect(typeof formikComponent.props.children).toBe("function");
  });

  // TextInputSignupScreen
  describe("TextInputSingupScreen", () => {
    let name;
    let emailInput;
    let dateOfBirth;
    let passwordInput;
    let confirmPasswordInput;

    const renderForm = () => {
      const { getByTestId } = signupScreenRender;
      name = getByTestId("name");
      emailInput = getByTestId("signup-screen-email-input");
      dateOfBirth = getByTestId("date-of-birth");
      passwordInput = getByTestId("signup-screen-password-input");
      confirmPasswordInput = getByTestId("confirm-password-input");
    };

    beforeEach(() => {
      renderForm();
    });

    afterEach(() => {
      cleanup();
    });

    describe("Rendering", () => {
      test("Renders correctly the name", () => {
        expect(name).toBeTruthy();
      });

      test("Renders correctly the email-input", () => {
        expect(emailInput).toBeTruthy();
      });

      test("Renders correctly the date-of-birth", () => {
        expect(dateOfBirth).toBeTruthy();
      });

      test("Renders correctly the password-input", () => {
        expect(passwordInput).toBeTruthy();
      });

      test("Renders correctly the confirm-password-input", () => {
        expect(confirmPasswordInput).toBeTruthy();
      });
    });
  });

  describe("DateTimePicker", () => {
    let dateOfBirth;

    const renderForm = () => {
      const { getByTestId } = signupScreenRender;
      dateOfBirth = getByTestId("date-of-birth");
    };

    beforeEach(() => {
      renderForm();
    });

    afterEach(() => {
      cleanup();
    });

    test("Renders correctly the date-of-birth input field", () => {
      expect(dateOfBirth).toBeTruthy();
    });

    test("Show DateTimePicker after clicking on the date-of-birth input field", async () => {
      act(() => {
        fireEvent.press(dateOfBirth);
      });

      await waitFor(() => {
        const dateTimePicker =
          signupScreenRender.queryByTestId("date-time-picker");
        expect(dateTimePicker).toBeTruthy();
      });
    });

    test("Change date in DateTimePicker", async () => {
      act(() => {
        fireEvent.press(dateOfBirth);
      });

      await waitFor(() => {
        const dateTimePicker =
          signupScreenRender.queryByTestId("date-time-picker");
        expect(dateTimePicker).toBeTruthy();

        act(() => {
          fireEvent(dateTimePicker, "onChange", {
            nativeEvent: { timestamp: "Tue Feb 01 2022" }
          });
        });
      });

      await waitFor(() => {
        expect(dateOfBirth.props.value).toBe("Tue Feb 01 2022");
      });
    });
  });

  // StyledButton
  describe("StyledButton", () => {
    afterEach(() => {
      cleanup();
    });

    test("Render StyledButton", () => {
      const { getByTestId } = signupScreenRender;
      const styledButtonElement = getByTestId("signup-styled-button");
      expect(styledButtonElement).toBeTruthy();
    });
  });

  // MsgBox
  describe("MsgBox", () => {
    afterEach(() => {
      cleanup();
    });

    test("Render a MsgBox", () => {
      const { getByTestId } = signupScreenRender;
      const msgBoxElement = getByTestId("msg-box");
      expect(msgBoxElement).toBeTruthy();
    });

    test("MsgBox should render a string of letters, numbers or spaces", async () => {
      const { getByTestId } = signupScreenRender;
      const msgBoxElement = getByTestId("msg-box");
      const textContent = msgBoxElement.props.children.toString();
      expect(textContent).toMatch(/^[a-zA-Z0-9.\s]*$/);
    });
  });

  describe("FooterView", () => {
    afterEach(() => {
      cleanup();
    });

    test("Render correctly", () => {
      const { getByTestId } = signupScreenRender;
      const FooterViewElement = getByTestId("footer-view");
      expect(FooterViewElement).toBeTruthy();
    });
  });

  describe("FooterText", () => {
    afterEach(() => {
      cleanup();
    });

    test("Render correctly", () => {
      const { getByTestId } = signupScreenRender;
      const footerTextElement = getByTestId("signup-footer-text");
      expect(footerTextElement).toBeTruthy();
    });

    test("Render the correct text", () => {
      const { getByTestId } = signupScreenRender;
      const footerTextElement = getByTestId("signup-footer-text");
      const textContent = footerTextElement.props.children.toString();
      expect(textContent).toMatch("Already have an account?");
    });
  });
});

// Navigation Tests
describe("Navigation Tests", () => {
  test("Navigate to Login Screen", () => {
    const mockNav = { navigate: jest.fn() };
    const { getByTestId } = render(
      <Provider store={store}>
        <CredentialsContext.Provider
          value={{ storedCredentials: null, setStoredCredentials: jest.fn() }}>
          <SignupScreen navigation={mockNav} />
        </CredentialsContext.Provider>
      </Provider>
    );

    const loginText = getByTestId("footer-login-link");
    fireEvent.press(loginText);
    expect(mockNav.navigate).toHaveBeenCalledWith("LoginScreen");
  });
});
