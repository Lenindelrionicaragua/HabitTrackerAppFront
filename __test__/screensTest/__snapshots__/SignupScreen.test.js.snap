// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupScreen Renders correctly the SignupScreen Component 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
  testID="keyboard-avoiding-view"
>
  <RCTScrollView
    testID="scroll-view"
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="pressable"
      >
        <RCTSafeAreaView
          style={
            {
              "backgroundColor": "#f9f7f3",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
              "width": "100%",
            }
          }
          testID="styled-container"
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
                "width": "100%",
              }
            }
            testID="inner-container"
          >
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "textAlign": "center",
                }
              }
              testID="page-title"
            >
              ZenTimer
            </Text>
            <Text
              style={
                {
                  "color": "#1a1204",
                  "fontSize": 10,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "marginBottom": 20,
                }
              }
              testID="sub-title"
            >
              Account Sign Up
            </Text>
            <View
              style={
                {
                  "width": "90%",
                }
              }
            >
              <View
                testID="text-input-signup-screen"
              >
                <View
                  style={
                    {
                      "left": 15,
                      "position": "absolute",
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="left-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "octicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "color": "#1a1204",
                      "fontSize": 13,
                      "textAlign": "left",
                    }
                  }
                >
                  Name
                </Text>
                <TextInput
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Zen User"
                  placeholderTextColor="#f9f7f3"
                  style={
                    {
                      "backgroundColor": "#ffc500",
                      "borderBottomLeftRadius": 5,
                      "borderBottomRightRadius": 5,
                      "borderTopLeftRadius": 5,
                      "borderTopRightRadius": 5,
                      "color": "#1a1204",
                      "fontSize": 16,
                      "height": 60,
                      "marginBottom": 10,
                      "marginVertical": 3,
                      "paddingBottom": 15,
                      "paddingLeft": 55,
                      "paddingRight": 55,
                      "paddingTop": 15,
                    }
                  }
                  testID="name"
                  value=""
                />
              </View>
              <View
                testID="text-input-signup-screen"
              >
                <View
                  style={
                    {
                      "left": 15,
                      "position": "absolute",
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="left-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "octicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "color": "#1a1204",
                      "fontSize": 13,
                      "textAlign": "left",
                    }
                  }
                >
                  Email Address
                </Text>
                <TextInput
                  keyboardType="email-address"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="serenity@gmail.com"
                  placeholderTextColor="#f9f7f3"
                  style={
                    {
                      "backgroundColor": "#ffc500",
                      "borderBottomLeftRadius": 5,
                      "borderBottomRightRadius": 5,
                      "borderTopLeftRadius": 5,
                      "borderTopRightRadius": 5,
                      "color": "#1a1204",
                      "fontSize": 16,
                      "height": 60,
                      "marginBottom": 10,
                      "marginVertical": 3,
                      "paddingBottom": 15,
                      "paddingLeft": 55,
                      "paddingRight": 55,
                      "paddingTop": 15,
                    }
                  }
                  testID="email-input"
                  value=""
                />
              </View>
              <View
                testID="text-input-signup-screen"
              >
                <View
                  style={
                    {
                      "left": 15,
                      "position": "absolute",
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="left-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "octicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "color": "#1a1204",
                      "fontSize": 13,
                      "textAlign": "left",
                    }
                  }
                >
                  Date of Birth
                </Text>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <TextInput
                    editable={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="Tue Feb 01 1984"
                    placeholderTextColor="#f9f7f3"
                    style={
                      {
                        "backgroundColor": "#ffc500",
                        "borderBottomLeftRadius": 5,
                        "borderBottomRightRadius": 5,
                        "borderTopLeftRadius": 5,
                        "borderTopRightRadius": 5,
                        "color": "#1a1204",
                        "fontSize": 16,
                        "height": 60,
                        "marginBottom": 10,
                        "marginVertical": 3,
                        "paddingBottom": 15,
                        "paddingLeft": 55,
                        "paddingRight": 55,
                        "paddingTop": 15,
                      }
                    }
                    testID="date-of-birth"
                    value=""
                  />
                </View>
              </View>
              <View
                testID="text-input-signup-screen"
              >
                <View
                  style={
                    {
                      "left": 15,
                      "position": "absolute",
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="left-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "octicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "color": "#1a1204",
                      "fontSize": 13,
                      "textAlign": "left",
                    }
                  }
                >
                  Password
                </Text>
                <TextInput
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="* * * * * * *"
                  placeholderTextColor="#f9f7f3"
                  secureTextEntry={true}
                  style={
                    {
                      "backgroundColor": "#ffc500",
                      "borderBottomLeftRadius": 5,
                      "borderBottomRightRadius": 5,
                      "borderTopLeftRadius": 5,
                      "borderTopRightRadius": 5,
                      "color": "#1a1204",
                      "fontSize": 16,
                      "height": 60,
                      "marginBottom": 10,
                      "marginVertical": 3,
                      "paddingBottom": 15,
                      "paddingLeft": 55,
                      "paddingRight": 55,
                      "paddingTop": 15,
                    }
                  }
                  testID="password-input"
                  value=""
                />
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "position": "absolute",
                      "right": 15,
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="right-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#f9f7f3",
                          "fontSize": 25,
                        },
                        undefined,
                        {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <View
                testID="text-input-signup-screen"
              >
                <View
                  style={
                    {
                      "left": 15,
                      "position": "absolute",
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="left-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "octicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "color": "#1a1204",
                      "fontSize": 13,
                      "textAlign": "left",
                    }
                  }
                >
                  Confirm Password
                </Text>
                <TextInput
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="* * * * * * *"
                  placeholderTextColor="#f9f7f3"
                  secureTextEntry={true}
                  style={
                    {
                      "backgroundColor": "#ffc500",
                      "borderBottomLeftRadius": 5,
                      "borderBottomRightRadius": 5,
                      "borderTopLeftRadius": 5,
                      "borderTopRightRadius": 5,
                      "color": "#1a1204",
                      "fontSize": 16,
                      "height": 60,
                      "marginBottom": 10,
                      "marginVertical": 3,
                      "paddingBottom": 15,
                      "paddingLeft": 55,
                      "paddingRight": 55,
                      "paddingTop": 15,
                    }
                  }
                  testID="confirm-password-input"
                  value=""
                />
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "position": "absolute",
                      "right": 15,
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="right-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#f9f7f3",
                          "fontSize": 25,
                        },
                        undefined,
                        {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <Text
                style={
                  {
                    "color": "#ff0000",
                    "fontSize": 13,
                    "textAlign": "center",
                  }
                }
                testID="msg-box"
                type="ERROR"
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#000000",
                    "borderBottomLeftRadius": 5,
                    "borderBottomRightRadius": 5,
                    "borderTopLeftRadius": 5,
                    "borderTopRightRadius": 5,
                    "height": 60,
                    "justifyContent": "center",
                    "marginVertical": 5,
                    "paddingBottom": 15,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 15,
                  }
                }
                testID="login-styled-button"
              >
                <Text
                  style={
                    {
                      "color": "#ffe1a8",
                      "fontSize": 16,
                    }
                  }
                  testID="login-button-text"
                >
                  Sign Up
                </Text>
              </View>
              <View
                style={
                  {
                    "backgroundColor": "#ffc500",
                    "height": 1,
                    "marginVertical": 10,
                    "width": "100%",
                  }
                }
                testID="line"
              />
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
                testID="footer-view"
              >
                <Text
                  style={
                    {
                      "alignContent": "center",
                      "color": "#1a1204",
                      "fontSize": 15,
                      "justifyContent": "center",
                    }
                  }
                  testID="footer-text"
                >
                  Already have an account?
                </Text>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    }
                  }
                  testID="footer-login-link"
                >
                  <Text
                    style={
                      {
                        "color": "#ffc554",
                        "fontSize": 15,
                      }
                    }
                    testID="footer-login-link-content"
                  >
                    Login
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
